### PR TITLE
Aerothermal example

### PR DESCRIPTION
**Pull request by joanibal**
This PR adds the aerothermal coupling group and scenario. 
It also adds an aerothermal example.

### CODE REVIEW

**Comment by joanibal** [comment]
Here are the related PRs.
[FuntoFem](https://github.com/smdogroup/funtofem/pull/106)
[tacs](https://github.com/smdogroup/tacs/pull/176

The modification to the ADflow repo have not been made on the main branch. 
But you can find them here on (myfork/dev)[https://github.com/joanibal/adflow/tree/dev].
There will be a PR for that too eventually, but I wanted to get the rest out there so @friedenhe could use it.

File: mphys/multipoint.py [hunk 1]
```diff
@@ -69,7 +69,11 @@ def mphys_connect_scenario_coordinate_source(self, source, scenarios, discipline
             for discipline in disciplines_list:
                 src = f'{source}.x_{discipline}0'
                 target = f'{scenario}.x_{discipline}0'
+                print(f'Connecting {src} to {target}')
                 self.connect(src, target)
+        
+        import pdb
```
Comment (friedenhe) [inline mphys/multipoint.py:75]: Is this for debugging? Maybe we want to remove it.

File: mphys/mphys_group.py [hunk 2]
```diff
@@ -31,12 +31,14 @@ def configure(self):
     def _mphys_promote_by_tag(self, iotype, tag):
         for subsystem in self.mphys_subsystems:
             promoted = []
+            print(f'promoting from {subsystem.name}')
```
Comment (friedenhe) [inline mphys/mphys_group.py:34]: Is this for debugging? I suggest we either remove it or add an if statement to print the information from the root proc only.

File: mphys/scenario_aerothermal.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        thermalxfer_builder = self.options['thermalxfer_builder']
+        geometry_builder = self.options['geometry_builder']
+
+        if self.options['in_MultipointParallel']:
+            self._mphys_initialize_builders(aero_builder, thermal_builder,
+                                            thermalxfer_builder, geometry_builder)
+            self._mphys_add_mesh_and_geometry_subsystems(aero_builder, thermal_builder,
+                                                         geometry_builder)
+
+        self.mphys_add_pre_coupling_subsystem('aero', aero_builder, self.name)
```
Comment (friedenhe) [inline mphys/scenario_aerothermal.py:37]: It seems that this function name is deprecated, and now it is called `_mphys_add_pre_coupling_subsystem_from_builder` in scenario.py. Please double-check.

File: mphys/multipoint.py (hunk 1 above)
Comment (joanibal) [reply mphys/multipoint.py]: good catch.

File: mphys/mphys_group.py (hunk 2 above)
Comment (joanibal) [reply mphys/mphys_group.py:34]: I have removed this

File: mphys/scenario_convection.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+                self.add_subsystem('mesh',aero_builder.get_mesh_coordinate_subsystem(self.name))
+                self.mphys_add_subsystem('geometry',geometry_builder.get_mesh_coordinate_subsystem(self.name))
+                self.connect('mesh.x_aero0','geometry.x_aero_in')
+            else:
+                self.mphys_add_subsystem('mesh',aero_builder.get_mesh_coordinate_subsystem(self.name))
+            self.connect('x_aero0','x_aero')
+
+        self.mphys_add_pre_coupling_subsystem('aero', aero_builder, self.name)
+        self.mphys_add_subsystem('coupling',aero_builder.get_coupling_group_subsystem(self.name))
+        self.mphys_add_post_coupling_subsystem('aero', aero_builder, self.name)
```
Comment (kejacobson) [inline mphys/scenario_convection.py:35]: I'm not seeing how this `scenario_convection.py` is different than the `scenario_aerodynamic.py` currently in mphys?

File: mphys/scenario_aerothermal.py (hunk 3 above)
Comment (kejacobson) [reply mphys/scenario_aerothermal.py:37]: Yes, Ping is correct. This change was to allow user-defined subsystems to be added to scenarios in the base class' `setup` method.

File: mphys/mphys_group.py [hunk 5]
```diff
@@ -31,12 +31,14 @@ def configure(self):
...
             tagged_variables = subsystem.get_io_metadata(iotypes=iotype,
                                                          metadata_keys=['tags'],
                                                          tags=tag)
             for val in tagged_variables.values():
                 variable = val['prom_name']
                 if variable not in promoted:
+                    print('promoting var:', variable)
```
Comment (kejacobson) [inline mphys/mphys_group.py:41]: another debug print here

**Review by kejacobson — CHANGES_REQUESTED** [review]
Looks good overall, just a few small changes to remove the other debug print, move to the new method for physics-specific scenario setup, and maybe remove the convention scenario in favor of the existing aerodynamics scenario

File: mphys/mphys_group.py (hunk 5 above)
Comment (kejacobson) [reply mphys/mphys_group.py:41]: Will be addressed by @friedenhe in a subsequent PR

**Review by kejacobson — APPROVED** [review]

**Pull request merged**